injectproxy/routes.go

```diff
@@ -256,24 +257,49 @@ func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 	})
 }
 
-// getLabelValue returns the statically set label value, or the label value
-// sent through a URL parameter.
-// It returns an error when either the value is found in both places, or is not found at all.
+// getLabelValue returns a regular expression concatenated using "|" by the statically set label value,
+// or by the label value sent through a URL parameter.
+// It returns an error when the URL parameter cannot be parsed,
+// or the value is found in both places, or is not found at all,
+// or one of the provided parameters is empty
 func (r *routes) getLabelValue(req *http.Request) (string, error) {
-	formValue := req.FormValue(r.label)
-	if r.labelValue != "" && formValue != "" {
+	err := req.ParseForm()
+	if err != nil {
+		return "", fmt.Errorf("the query parameter can not be parsed: %w", err)
+	}
+
+	formValues := req.Form[r.label]
+
+	if len(r.labelValue) > 0 && len(formValues) > 0 {
 		return "", fmt.Errorf("a static value for the %s label has already been specified", r.label)
 	}
 
-	if r.labelValue == "" && formValue == "" {
+	if len(r.labelValue) == 0 && len(formValues) == 0 {
 		return "", fmt.Errorf("the %q query parameter must be provided", r.label)
 	}
 
-	if r.labelValue != "" {
-		return r.labelValue, nil
+	if len(r.labelValue) > 0 {
+		return r.joinMultipleLabelValues(r.labelValue), nil
+	}
+
+	// Check if every value is non-empty
+	for i := range formValues {
+		if formValues[i] == "" {
+			return "", fmt.Errorf("the %q query parameter must be provided", r.label)
+		}
+	}
+
+	return r.joinMultipleLabelValues(formValues), nil
+}
+
+func (r *routes) joinMultipleLabelValues(labelValues []string) string {
+	for i := range labelValues {
+		labelValues[i] = regexp.QuoteMeta(labelValues[i])
 	}
```

Comment:
this would modify the given slice which we don't want, especially in the case of the static label values.

```suggestion
        lvs := make([]string, len(labelValues)
	for i := range labelValues {
		lvs[i] = regexp.QuoteMeta(labelValues[i])
	}
```